pkg/kubelet/volume/iscsi_pd/iscsi_pd.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iscsi_pd
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/pd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/golang/glog"
+)
+
+// This is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.Plugin {
+	return []volume.Plugin{&ISCSIDiskPlugin{nil, false, "", "", ""},
+		&ISCSIDiskPlugin{nil, true, "", "", ""}}
+}
+
+type ISCSIDiskPlugin struct {
+	host       volume.Host
+	legacyMode bool // if set, plugin answers to the legacy name
+	portal     string
+	iqn        string
+	lun        string
+}
+
+var _ volume.Plugin = &ISCSIDiskPlugin{}
+
+const (
+	ISCSIDiskPluginName       = "kubernetes.io/iscsi-pd"
+	ISCSIDiskPluginLegacyName = "iscsi-pd"
+)
+
+func (plugin *ISCSIDiskPlugin) Init(host volume.Host) {
+	plugin.host = host
+}
+
+func (plugin *ISCSIDiskPlugin) Name() string {
+	if plugin.legacyMode {
```

Comment:
What is legacyMode for this new volume?   I see this in gce_pd, but I assume this is because that implementation is one of the earlier volumes.  I assume there are no legacy iSCSCI disks to handle.  

That's two assumptions that might bite me in the rear.